test/test/otlp/OtlpSignalCorrelationTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package test.otlp;
+
+import one.profiler.AsyncProfiler;
+import one.profiler.TraceContext;
+import io.opentelemetry.proto.profiles.v1development.*;
+import java.time.Duration;
+import java.time.Instant;
+import com.google.protobuf.ByteString;
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import java.util.Optional;
+
+public class OtlpSignalCorrelationTest {
+    private static final Duration TEST_DURATION = Duration.ofSeconds(1);
+
+    private static final String TRACE_ID_1 = "09a6c61f1181ce4fc439e5728c5fef75";
+    private static final String SPAN_ID_1 = "b98c89ad5d208dcc";
+    private static final String TRACE_ID_1_ALT = "55812e1a0e7d80817be621dedec6accb";
+    private static final String SPAN_ID_1_ALT = "ef95eccf36472d99";
+    private static final String TRACE_ID_2 = "adf1a04340a7a3995571db46c3c648dc";
+    private static final String SPAN_ID_2 = "86dd157fc2677b88";
+    private static final String TRACE_ID_3 = "761d3a2cbae70d2e625985c887874266";
+    private static final String SPAN_ID_3 = "839dfc1575e1ea94";
+
+    public static void main(String[] args) throws Exception {
+        AsyncProfiler profiler = AsyncProfiler.getInstance();
+        profiler.execute("start,otlp,threads");
+
+        Thread[] threads = new Thread[3];
+        
+        threads[0] = new Thread(() -> {
+            TraceContext.setTraceContext(TRACE_ID_1, SPAN_ID_1);
+            burnCpu();
+            TraceContext.setTraceContext(TRACE_ID_1_ALT, SPAN_ID_1_ALT);
+            burnCpu();
+        }, "CorrelationThread1");
+
+        threads[1] = new Thread(() -> {
+            TraceContext.setTraceContext(TRACE_ID_2, SPAN_ID_2);
+            burnCpu();
+        }, "CorrelationThread2");
+
+        threads[2] = new Thread(() -> {
+            TraceContext.setTraceContext(TRACE_ID_3, SPAN_ID_3);
+            burnCpu();
+        }, "CorrelationThread3");
+
+        for (Thread t : threads) {
+            t.start();
+        }
+        
+        for (Thread t : threads) {
+            t.join();
+        }
+
+        ProfilesData data = dumpAndGetProfile(profiler);
+        profiler.stop();
+        
+        assert data.getDictionary().getLinkTableCount() == 5 : 
+            data.getDictionary().getLinkTableCount();
```

Comment:
Please add some context here, otherwise we'll get `AssertionError + someNumber` which is not really informative